### ISSUE TITLE
kube-state-metrics is giving ApiVersion deprecation warning even if deprecated ApiVersion is not deployed

### ISSUE THREAD

**Issue by ozkansari37** [kubernetes/kube-state-metrics#1661]
**What happened**:
kube-state-metrics prints logs like below on my newly created EKS Cluster v1.21, but I don't have any components with those ApiVersions. 
```
W0103 15:18:33.359981       1 warnings.go:70] batch/v1beta1 CronJob is deprecated in v1.21+, unavailable in v1.25+; use batch/v1 CronJob
W0103 15:18:33.362190       1 warnings.go:70] batch/v1beta1 CronJob is deprecated in v1.21+, unavailable in v1.25+; use batch/v1 CronJob
W0103 15:18:33.365912       1 warnings.go:70] policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
W0103 15:18:33.374825       1 warnings.go:70] policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
```

**What you expected to happen**:
This warning shouldn't be seen because since I don't have any resources with those ApiVersion, it makes no sense I guess.

**How to reproduce it (as minimally and precisely as possible)**:
```
(On AWS EKS v1.21)
git clone https://github.com/kubernetes/kube-state-metrics.git
kubectl apply -f kube-state-metrics/examples/standard/
kubectl logs -f <pod-name> -n kube-system
```

**Anything else we need to know?**:
When I check the metrics with something like, 
```
kubectl port-forward svc/kube-state-metrics 30135:8080 -n kube-system
http://127.0.0.1:30135/metrics
```
Metrics is being shown, so I assume that kube-state-metrics is working properly. But I don't understand why I'm still getting this warnings.

**Environment**:
- kube-state-metrics version: 
```
v2.3.0
```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:10:45Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-eks-06eac09", GitCommit:"5f6d83fe4cb7febb5f4f4e39b3b2b64ebbbe3e97", GitTreeState:"clean", BuildDate:"2021-09-13T14:20:15Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
```
AWS EKS 1.21
```
Any help will be appreciated. Thank you.

**Comment by slashpai** [comment]
I see we use this apiVersion in
 https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/cronjob.go

https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/poddisruptionbudget.go

I would wait for @fpetkovski to confirm, I can work on the fix

**Comment by fpetkovski** [comment]
I believe there's already a PR to address this: https://github.com/kubernetes/kube-state-metrics/pull/1491. Now that 1.23 is out, we can rebase and merge it.

**Comment by ozkansari37** [comment]
Thank you for the effort, it would be really appreciated @fpetkovski  :)

**Comment by fpetkovski** [comment]
Closed via #1491
